src/Acuminator/Acuminator.Analyzers/StaticAnalysis/NonPublicExtensions/NonPublicGraphAndDacExtensionAnalyzer.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Acuminator.Analyzers.StaticAnalysis.Dac;
+using Acuminator.Analyzers.StaticAnalysis.PXGraph;
+using Acuminator.Utilities.DiagnosticSuppression;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Acuminator.Utilities.Roslyn.Semantic.Dac;
+using Acuminator.Utilities.Roslyn.Semantic.PXGraph;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Acuminator.Analyzers.StaticAnalysis.NonPublicExtensions
+{
+	/// <summary>
+	/// An analyzer that checks that DAC and graph extensions are public.
+	/// </summary>
+	public class NonPublicGraphAndDacExtensionAnalyzer : IDacAnalyzer, IPXGraphAnalyzer
+	{
+		public ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(Descriptors.PX1022_NonPublicDacExtension,
+								  Descriptors.PX1022_NonPublicGraphExtension);
+
+		bool IDacAnalyzer.ShouldAnalyze(PXContext pxContext, DacSemanticModel dac) =>
+			dac?.DacType == DacType.DacExtension;
+
+		bool IPXGraphAnalyzer.ShouldAnalyze(PXContext pxContext, PXGraphSemanticModel graph) =>
+			graph?.Type == GraphType.PXGraphExtension;
+
+		void IDacAnalyzer.Analyze(SymbolAnalysisContext context, PXContext pxContext, DacSemanticModel dacExtension) =>
+			CheckSymbolIsPublic(context, pxContext, dacExtension);
+
+		void IPXGraphAnalyzer.Analyze(SymbolAnalysisContext context, PXContext pxContext, PXGraphSemanticModel graphExtension) =>
+			CheckSymbolIsPublic(context, pxContext, graphExtension);
+
+		private void CheckSymbolIsPublic(SymbolAnalysisContext context, PXContext pxContext, ISemanticModel extension)
```

Comment:
Yes, because the partial is more like a syntax concept and the check for public works on symbols, which is a semantic data and is a level above the syntax - symbols already have all partial declarations combined, and the info about accessibility can be retrieved from them.

Reply:
@dropsonic  by the way the code fix for this diagnostic doesn't "fix" partial type declarations. It fix only the type declaration with alert, and if the type happens to be a partial type (which is a rare case for extensions but still possible) then it leaves the code with a compile error that the user will have to fix himself.
I think it's ok because:
1) scenario is really rare.
2) Code fix that fixes different documents in solution is much more complex. We don't have any such code fixes in Acuminator currently
3) It's impossible to leave it unattended - the user will have to fix it due to compilation errors